deploy/terraform-gcp/main.tf

```diff
@@ -0,0 +1,168 @@
+provider "google" {
+  project = "excellent-ion-277200"
+  region  = var.region
+  zone    = var.zone
+}
+
+resource "random_id" "password" {
+  byte_length = 16
+}
+
+locals {
+  password = format("%s", random_id.password.hex)
+}
+
+resource "google_container_cluster" "cluster" {
+  name = var.cluster-name
+
+  # Create a zonal cluster. In production, you want a regional cluster with
+  # multiple masters spread across zones in the region.
+  location        = var.zone
+
+  # We can't create a cluster with no node pool defined, but we want to only use
+  # separately managed node pools. So we create the smallest possible default
+  # node pool and immediately delete it.
+  remove_default_node_pool = true
+  initial_node_count       = 1
+
+  master_auth {
+    username = "client"
+    password = local.password
+
+    client_certificate_config {
+      issue_client_certificate = true
+    }
+  }
+
+  ip_allocation_policy {
+    cluster_ipv4_cidr_block   = var.pod-cidr
+    services_ipv4_cidr_block  = var.service-cidr
+  }
+
+  master_authorized_networks_config {
+    cidr_blocks {
+      # Limit this in production to your client IPs.
+      cidr_block = "0.0.0.0/0"
+    }
+  }
+}
+
+resource "google_container_node_pool" "node-pool" {
+  name       = "node-pool-${var.cluster-name}"
+  location   = var.zone
+  cluster    = google_container_cluster.cluster.name
+  node_count = 1
+
+  node_config {
+    preemptible  = false
+    machine_type = "n1-standard-1"
+
+    metadata = {
+      disable-legacy-endpoints = "true"
+    }
+
+    oauth_scopes = [
+      "https://www.googleapis.com/auth/logging.write",
+      "https://www.googleapis.com/auth/monitoring",
+    ]
+  }
+}
+
+resource "google_filestore_instance" "filestore" {
```

Comment:
Do we want to include filestore along with this manifest?  I was under the impression that would be an enterprisey add-on.

Reply:
Yes, this is just a few lines, nothing fancy. The business logic for image caching will be in a separate controller and itzo add-on.